lib/lightning/version_control/version_control.ex

```diff
@@ -35,6 +37,10 @@ defmodule Lightning.VersionControl do
 
     Repo.transact(fn ->
       with {:ok, repo_connection} <- Repo.insert(changeset),
+           {:ok, _audit} <-
+             repo_connection
+             |> Audit.repo_connection(:created, user)
+             |> Repo.insert(),
```

Comment:
I see it wasn't using the Multi here and the `Audit.derive_events` depends on it. Currently the signature is 
```
  def derive_events(multi, changeset, user) do
    [:history_retention_period, :dataclip_retention_period] 
```
and the PR introduces more two event types here:
https://github.com/OpenFn/lightning/blob/09e555336c7c3497fe5f1a742538bb20374de52a/lib/lightning/projects/audit.ex#L8-L12

Just to confirm, would the reason to not standardise the use of `derive_events` be the tracking of two connection properties?

Reply:
@jyeshe I can't remember if this was a conscious decision - while we do have a changeset when creating a repo connection (and hence could use a changeset) we do not have a changeset when we are deleting the repo connection. 

Therefore, I may have gravitated early on to having these events be handled by a methodthat take a repo connection. Although, even looking at the code now, I am not sure if it is a candidate for generalisation :).

Happy to hear counter-opinions :)

Reply:
Humm @rorymckinley I see that even though it's a projects audit it's not a config on the project itself. As there is no change on the project it looks like as a candidate for `Lightning.VersionControl` to avoid mixing this two entities and making it clear which one is changed.

Reply:
@jyeshe So far, I have been using the 'subject' (or the item)  as the deciding factor ito who owns the Audit module responsible. For this event, the requirement is that the subject should be the project as the [issue](https://github.com/OpenFn/lightning/issues/2578) refers to `GitHub configuration for {project_name}` suggesting that the project is the subject of the event. (Unfortunately, I can't see a way to link you directly to the relevant section of the page, so I a afraid you will need to search for "Github Integration").

I think you are correct from a technical POV wrt this being a `VersionControl` event, but I think from a user perspective it makes sense to link it to the project as that is what is affected in userland? I am happy to move it to VersionControl if you can convince Ayo to change the spec :).

Reply:
I think the text could still point to a project, be associated to a project, but be more specific that it's not a project attribute that is being changed. Even so, if we don't want to change the label to the user, at least could we have an audit for `VersionControl` implementation/model wise?

It would prevent someone having the same question for a generalisation refactoring and more importantly make it clear that it's not auditing Project settings. What the `moduledoc` says was my initial "bias" on the begging of the review for things on `Lightning.Projects.Audit`: _Generate Audit changesets for selected changes to project settings_.
 
For a auditing changes on GitHub ("project_repo_connections", not on "projects") I think we should have:
```
defmodule Lightning.VersionControl.Audit do
  @moduledoc """
  Generate Audit changesets for GitHub connection settings changes.
  """
  use Lightning.Auditing.Audit,
    repo: Lightning.Repo,
    item: "project" # if not "version control",
    events: [
      "repo_connection_created",
      "repo_connection_removed"
    ]
  """
```

Reply:
@rorymckinley haven't tag you before, this is a change I believe we should make.